src/Documenter.jl

```diff
@@ -441,4 +445,93 @@ end
 
 end
 
+"""
+    generate(
+        pkgname;
+        dir = "<package directory>"
+    )
+
+Creates a documentation stub for a package called `pkgname`. The location of
+the documentation is assumed to be `<package directory>/docs`, but this can
+be overriden with keyword arguments.
+
+It creates the following files
+
+    docs/
+        .gitignore
+        src/index.md
+        make.jl
+        mkdocs.yml
+
+# Positionals
+
+**`pkgname`** is the name of the package (without `.jl`). It is used to
+determine the location of the documentation if `dir` is not provided.
+
+# Keywords
+
+**`dir`** defines the directory where the documentation will be generated.
+It defaults to `<package directory>/docs`. The directory must not exist.
+
+# Examples
+
+    julia> using Documenter
+
+    julia> Documenter.generate("MyPackageName")
+    [ ... output ... ]
+
+"""
+function generate(pkgname::AbstractString; dir=nothing)
+    # TODO:
+    #   - `format` argument to determine the deployment (e.g. mkdocs)
+    #   - set up deployment to `gh-pages`?
```

Comment:
Sounds reasonable, perhaps as a subsequent PR, since that would need some changes to the hosting docs as well.